crypto/http/http_lib.c

```diff
@@ -114,3 +120,50 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
     OPENSSL_free(buf);
     return 0;
 }
+
+int http_use_proxy(const char *no_proxy, const char *server)
+{
+    size_t sl;
+    const char *found = NULL;
+
+    if (!ossl_assert(server != NULL))
+        return 0;
+    sl = strlen(server);
+
+    if (no_proxy == NULL)
+        no_proxy = getenv("no_proxy");
+    if (no_proxy == NULL)
+        no_proxy = getenv(OPENSSL_NO_PROXY);
```

Comment:
I had actually been thinking that the environment variable itself (i.e. the string) should be called `OPENSSL_NO_PROXY` - but what I hand't appreciated was that these names were deliberately chosen for compatibility with other software and aren't OpenSSL specific. Given that it might be worth a comment explaining that. This also seems like a reasonable justification for accepting the lowercase variant where we don't do that anywhere else for OpenSSL environment variables.

Reply:
I see.
I've added respective source comments to `http_lib.c` and a corresponding NOTES section to `OSSL_HTTP_transfer.pod`.